### PR TITLE
Set up language client treeview

### PR DESCRIPTION
**Pull request by kawwong**

### CODE REVIEW

File: client/src/models/ConditionTreeItem.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  get description (): string {
+    return this.isMet ? '' : 'Not met';
+  }
+
+  get tooltip (): string {
+    return this.label;
+  }
+
+  get iconPath (): string {
+    return join(this.rootPath, 'client', 'res', 'icons', `${this.isMet ? 'tracked' : 'notTracked'}.svg`);
```
Comment (cn894) [inline client/src/models/ConditionTreeItem.ts]: I would recommend taking the `${this.isMet ? 'tracked' : 'notTracked'}.svg` and putting it into a variable for readability.

File: client/src/utilities/treeItemUtilities.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+function generateIconPath (componentType: string, fulfilled: boolean, rootPath: string): string {
+  let iconPath = null;
+
+  if (componentType != null && fulfilled != null && rootPath != null) {
+    let iconKind = '';
+
+    if (!fulfilled) {
+      iconKind = 'Light';
+    }
+    iconPath = join(rootPath, 'client', 'res', 'icons', `${componentType + iconKind}.svg`);
```
Comment (cn894) [inline client/src/utilities/treeItemUtilities.ts]: I'd recommend moving the template string out into its own variable for readability, just like in the previous comment. Although this one is easier to read than the previous one.

File: client/src/models/MezzuriteDataProvider.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+class MezzuriteDataProvider implements TreeDataProvider<TreeItem> {
+  private _onDidChangeTreeData: EventEmitter<TreeItem> = new EventEmitter<TreeItem>();
+  readonly onDidChangeTreeData: Event<TreeItem> = this._onDidChangeTreeData.event;
+
+  constructor (
+    private components: MezzuriteComponent[],
+    private rootPath: string
+  ) {}
+
+  getChildren (element: MezzuriteTreeItem): Thenable<TreeItem[]> {
```
Comment (cn894) [inline client/src/models/MezzuriteDataProvider.ts]: Discussed offline. I think inline comments would be very helpful in explaining why the function is how it is.

File: client/src/utilities/treeItemUtilities.ts (hunk 2 above)
Comment (kawwong) [reply client/src/utilities/treeItemUtilities.ts]: Resolved via https://github.com/Microsoft/vscode-mezzurite/pull/33/commits/808368aeed44ebbcd418fe05e9542784865a8bc1.

File: client/src/models/ConditionTreeItem.ts (hunk 1 above)
Comment (kawwong) [reply client/src/models/ConditionTreeItem.ts]: Resolved via https://github.com/Microsoft/vscode-mezzurite/pull/33/commits/808368aeed44ebbcd418fe05e9542784865a8bc1.

File: client/src/models/MezzuriteDataProvider.ts (hunk 3 above)
Comment (kawwong) [reply client/src/models/MezzuriteDataProvider.ts]: Per the offline discussion, given the new direction of moving to javascript, the comparator was abstracted into a separate testable function in the future.

**Pull request merged**